### ISSUE TITLE
spawn_failed hook not being called

### ISSUE THREAD

**Issue by wpolicarpo** [phusion/passenger#2308]
## Issue report

**Question 1: What is the problem?**

We have a Rails application running under Passenger Enterprise + NGINX with 3 hooks configured: `detached_process`, `attached_process` and `spawn_failed`. The first 2 hooks are executed normally but `spawn_failed` seems to not be fired when the application fails to start.

I have both `passenger_rolling_restarts ` and `passenger_resist_deployment_errors` set to `on`. To simulate a `spawn_failed` situation, I force the application to fail during boot with a simple `raise "something"` in `config/application.rb` and restart the application with `passenger-config restart-app --rolling-restart /path/to/app`.

**Relevant NGINX config:**

```conf
passenger_ctl hook_detached_process /path/to/app/bin/log-passenger-event;
passenger_ctl hook_attached_process /path/to/app/bin/log-passenger-event;
passenger_ctl hook_spawn_failed     /path/to/app/bin/log-passenger-event;
```

**log-passenger-event**
```bash
echo $PASSENGER_HOOK_NAME >> /path/to/app/log/passenger.log

```
- What is the expected behavior?

There should have a log entry for `spawn_failed` because the application process failed to start.

- What is the actual behavior?

`spawn_failed` hook is never fired and no log entry is added.

**Question 2: Passenger version and integration mode**:

```
Version : 6.0.6 (Enterprise)
Instance: (nginx/1.14.0 Phusion_Passenger/6.0.6)
```

**Question 3: OS or Linux distro, platform (including version)**:

```
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.2 LTS"
$ uname -a
Linux MY_HOST 4.15.0-46-generic #49-Ubuntu SMP Wed Feb 6 09:33:07 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Question 4: Passenger installation method**:

[ ] RubyGems + Gemfile
[ ] RubyGems, no Gemfile
[x] Phusion APT repo
[ ] Phusion YUM repo
[ ] OS X Homebrew
[ ] source tarball
[ ] Other, please specify:

**Question 5: Your app's programming language (including any version managers) and framework (including versions)**:

- Ruby 2.6.3
- Rails 5.2.4.3

**Question 6: Are you using a PaaS and/or containerization? If so which one?**

No.

**Comment by CamJN** [comment]
We are aware of this issue and have a fix already lined up for 6.0.7. Thanks for the report though!

**Comment by CamJN** [comment]
6.0.7 is released
